test/js_test.go

```diff
@@ -1702,6 +1702,42 @@ func TestPurgeStream(t *testing.T) {
 	}
 }
 
+func TestStreamInfoSubjectInfo(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "foo",
+		Subjects: []string{"foo.*"},
+	})
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	if _, err := js.Publish("foo.A", []byte("")); err != nil {
+		t.Fatalf("Unexpected error during publish: %v", err)
+	}
+
```

Comment:
Having another publish on foo.B would help ensure that the library sends the proper filter, or alternatively, listen on the request subject to check that we are doing the right thing.
If you do publish on foo.B, you will need to adjust the rest of the test for the counts (NumSubjects should be 2), etc..

Reply:
I committed this by mistake along with the gomods. This was testing to see something Derek mentioned that I understood incorrectly.